switchlink/switchlink_link_test.cc

```diff
@@ -36,79 +36,96 @@ extern "C" {
-  DELETE_INTERFACE = 2,
-  CREATE_TUNNEL_INTERFACE = 3,
-  DELETE_TUNNEL_INTERFACE = 4,
+  CREATE_INTERFACE = 1,           // switchlink_create_interface
+  DELETE_INTERFACE = 2,           // switchlink_delete_interface
+  CREATE_TUNNEL_INTERFACE = 3,    // switchlink_create_tunnel_interface
+  DELETE_TUNNEL_INTERFACE = 4,    // switchlink_delete_tunnel_interface
+  CREATE_VRF = 5,                 // switchlink_create_vrf
 };
 
 /**
- * Result variables.
+ * Test results structure. Records the dummy function (handler) called and
+ * its parameter values.
  */
 struct test_results {
-  // Parameter values
-  switchlink_db_tunnel_interface_info_t tunnel_info;
-  switchlink_db_interface_info_t interface_info;
-  uint32_t ifindex;
+  union {
+    // Parameter values
+    switchlink_db_tunnel_interface_info_t tunnel_info;
+    switchlink_db_interface_info_t interface_info;
+    uint32_t ifindex;
+  };
   // Handler tracking
   enum handler_type handler;
-  int num_handler_calls;
 };
 
-static struct test_results results;
-
-
 /**
- * Dummy functions.
+ * Test results vector. Contains one entry for each dummy function called
+ * by the UUT.
  */
-void switchlink_create_interface(switchlink_db_interface_info_t *intf) {
-  results.handler = CREATE_INTERFACE;
-  results.num_handler_calls += 1;
+std::vector<test_results> results(2);
+
+//----------------------------------------------------------------------
+// Test doubles (dummy functions)
+//----------------------------------------------------------------------
+
+void switchlink_create_interface(switchlink_db_interface_info_t* intf) {
+  struct test_results temp = {0};
+  temp.handler = CREATE_INTERFACE;
   if (intf) {
-    results.interface_info = *intf;
+    temp.interface_info = *intf;
   }
+  results.push_back(temp);
 }
 
 void switchlink_delete_interface(uint32_t ifindex) {
-  results.handler = DELETE_INTERFACE;
-  results.num_handler_calls += 1;
-  results.ifindex = ifindex;
+  struct test_results temp = {0};
+  temp.handler = DELETE_INTERFACE;
+  temp.ifindex = ifindex;
+  results.push_back(temp);
 }
 
 void switchlink_create_tunnel_interface(
-    switchlink_db_tunnel_interface_info_t *tnl_intf) {
-  results.handler = CREATE_TUNNEL_INTERFACE;
-  results.num_handler_calls += 1;
+    switchlink_db_tunnel_interface_info_t* tnl_intf) {
+  struct test_results temp = {0};
+  temp.handler = CREATE_TUNNEL_INTERFACE;
   if (tnl_intf) {
-    results.tunnel_info = *tnl_intf;
+    temp.tunnel_info = *tnl_intf;
   }
+  results.push_back(temp);
 }
 
 void switchlink_delete_tunnel_interface(uint32_t ifindex) {
-  results.handler = DELETE_TUNNEL_INTERFACE;
-  results.num_handler_calls += 1;
-  results.ifindex = ifindex;
+  struct test_results temp = {0};
+  temp.handler = DELETE_TUNNEL_INTERFACE;
+  temp.ifindex = ifindex;
+  results.push_back(temp);
 }
 
 // fulfills an external reference
-int switchlink_create_vrf(switchlink_handle_t *vrf_h) {
+int switchlink_create_vrf(switchlink_handle_t* vrf_h) {
```

Comment:
Is there a reason for a `int` return for this function? Other test cases have `void`

Reply:
These are replacements for existing functions. They must conform to the signatures of the functions they replace.
The dummy function returns `int` because the function is replaces returns `int`.